### ISSUE TITLE
react/no-unused-state doesn't detect usage of `this.state` as an object

### ISSUE THREAD

**Issue by julienw** [yannickcr/eslint-plugin-react#1572]
Here is an example:

```javascript
<div
  className={classNames('overflowEdgeIndicator', className, this.state)}
>
```
`classNames` is the excellent [classnames package](https://www.npmjs.com/package/classnames).

Notice the use of `this.state` directly as a parameter to the `classNames` call. As a result, the rule `react/no-unused-state` doesn't detect we use some of the state properties and output an error.

**Comment by julienw** [comment]
thx !
